students/barb_matthews/lesson4/mr2.py

```diff
@@ -0,0 +1,141 @@
+
+prompt2 = "\n".join(("**********************************\nThank-You Options\n",
+                     "What do you want to do?:",
+                     "1) Enter a name",
+                     "2) List of donor names",
+                     "3) Return to Main Menu",
+                     "**********************************\n> "))
+
+def thanks():
+    """Prints a donor thank-you to the screen"""
+
+    while True:
+        choice = input(prompt2)
+        menu2.get(choice)()         # use menu dictionary to select correct function
+    return
+
+def report():
+    """Prints a report to the screen of donors and amounts"""
+    print("\n" * 100, "\a{:<23} | {:<15} | {:<15} | {:>15}".format("Name", "Total Donated ($)",
+                                                                   "Number of Donations",
+                                                                   "Average Amount ($)"))
+    print("-" * 87)
+
+    ## Sort the list of donors in descending total donation amount order
+
+    for each_key in donors:
+        #print('donors key', each_key, 'has value', donors[each_key])
+        total = sum(donors[each_key])
+        number = len(donors[each_key])
+        average = total/number
+        print("{:<24} | {:>17,.2f} | {:>19} | {:>15,.2f}".format(each_key, total, number, average))
+
+    print("\n\n")
+    return
+
+def exit_menu():
+    """Exits out of the interactive menu"""
+    print("Goodbye")
+    sys.exit()
+
+def thank_files():
+    """writes a donor thank-you and puts in a file for each donor"""
+    cwd = os.getcwd()       # Current working dir
+
+    for person, amount in donors.items():
+        pname = person.replace(' ', '_')
+        path = os.path.join(cwd, pname)
+        myfile = path + ".txt"
+        all = sum(donors[person])
+
+        with open(myfile, 'w') as f:
+            f.write("Thank you, %s, for your continuing generous donations of $%.2f.\n"
+                    "We appreciate your support. Sincerely, Grateful Team" % (person, all))
+            print("created file: ", myfile)
+
+    return
+
+def name_donors():
+    """Ask user for name of donor, add it if new, ask for dontation amount, print thanks.
+    If the name is found, just add the new donation to their record"""
+
+    name = input("What name?\n")
+    #print("you entered", name)
+    what = float(input("How much donated? --> "))
+
+    if name not in donors:
+        donors[name] = [what]
+        print("\n" * 50, "Dear %s,\n\nThank you for your generous donation of $%.2f. "
+                          "We appreciate your support.\n\nSincerely, Grateful Team" % (name, what))
+        print("\n" * 5)
+
+    else:
+        what_list = donors[name]
+        all = sum(what_list) + what
+        print("\n" * 50, "Thank you, %s for your continuing generous donations of $%.2f.\n"
+                          "We appreciate your support.\n\nSincerely, Grateful Team" % (name, all))
+        print("\n" * 5)
```

Comment:
@bapandaOK44 
when it is an existing donor you can see here that you're not actually updating donations for existing donor in here ^^

you would need to do something like `donors[name].append(what)`